### PR TITLE
Refactor queue file changes

### LINKED ISSUE #1754: [Refactor] queue_file_changes Azure App Function
Refactor the QueueFileChanges function and required sub classes to C#. Validate by disabling the python function and enabling the .net function in a testing environment before deploying the change to production.

**Blocked on templating**:
* https://github.com/microsoft/onefuzz/blob/447f1813b1c3fac0e932c575f51bbe1538194936/src/ApiService/ApiService/onefuzzlib/NotificationOperations.cs#L37
* https://github.com/microsoft/onefuzz/blob/447f1813b1c3fac0e932c575f51bbe1538194936/src/ApiService/ApiService/onefuzzlib/NotificationOperations.cs#L45
* https://github.com/microsoft/onefuzz/blob/447f1813b1c3fac0e932c575f51bbe1538194936/src/ApiService/ApiService/onefuzzlib/NotificationOperations.cs#L49

AB#39993385

AB#40263775

### PR DESCRIPTION
**Pull request by tevoinea**
## Summary of the Pull Request

_What is this about?_

Closes #1754 

## PR Checklist
* [ ] Applies to work item: #xxx
* [ ] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/onefuzz) and sign the CLI.
* [ ] Tests added/passed
* [ ] Requires documentation to be updated
* [ ] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #xxx

## Info on Pull Request

_What does this include?_

## Validation Steps Performed

_How does someone test & validate?_

### CODE REVIEW

File: src/ApiService/ApiService/onefuzzlib/Storage.cs [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+using System;
+using Azure.ResourceManager;
+using Azure.ResourceManager.Storage;
+using Azure.Core;
+using Microsoft.Extensions.Logging;
+using System.Text.Json;
+
+namespace Microsoft.OneFuzz.Service;
+
+public interface IStorage {
```
Comment (chkeita) [inline src/ApiService/ApiService/onefuzzlib/Storage.cs:11]: you can use StorageProvider.cs for this

File: src/ApiService/ApiService/QueueFileChanges.cs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    {
+        _logger = loggerFactory.CreateLogger<QueueFileChanges>();
+        _storageProvider = storageProvider;
+        _storage = storage;
+    }
+
+    [Function("QueueFileChanges")]
+    public async Task Run([QueueTrigger("file-changes-refactored", Connection = "AzureWebJobsStorage")] QueueMessage msg)
+    {
+        var fileChangeEvent = JsonSerializer.Deserialize<Dictionary<string, string>>(msg.Body.ToString());
```
Comment (chkeita) [inline src/ApiService/ApiService/QueueFileChanges.cs:33]: You will need the serializer option from EntityConverter for this operation

``` C#
var fileChangeEvent = JsonSerializer.Deserialize<Dictionary<string, string>>(msg, EntityConverter.GetJsonSerializerOptions());
```
```

File: src/ApiService/ApiService/onefuzzlib/Storage.cs (hunk 1 above)
Comment (tevoinea) [reply src/ApiService/ApiService/onefuzzlib/Storage.cs:11]: I was trying to do a direct mapping of this file: https://github.com/microsoft/onefuzz/blob/main/src/api-service/__app__/onefuzzlib/azure/storage.py

Comment (tevoinea) [reply src/ApiService/ApiService/onefuzzlib/Storage.cs:11]: I can move it to StorageProvider.cs if you still think it makes more sense to be there

Comment (chkeita) [reply src/ApiService/ApiService/onefuzzlib/Storage.cs:11]: That's ok you can keep  it this way. We will probably need a pass to unify both

File: src/ApiService/ApiService/Program.cs [hunk 3]
```diff
@@ -36,6 +36,8 @@ public static void Main()
         .ConfigureServices((context, services) =>
             services.AddSingleton(_ => new LogTracerFactory(GetLoggers()))
             .AddSingleton<IStorageProvider>(_ => new StorageProvider(EnvironmentVariables.OneFuzz.FuncStorage ?? throw new InvalidOperationException("Missing account id") ))
+            .AddSingleton<ICreds>(_ => new Creds())
```
Comment (chkeita) [inline src/ApiService/ApiService/Program.cs:39]: 👍

**Review by chkeita — APPROVED** [review]

**Pull request merged**